approval_polls/views.py

```
@@ -1,101 +1,212 @@
+
+        Arguments:
+            request (WSGIRequest): The request.POST member will contain all the data
+            from the form. This must be separated out and validated manually.
+
+            pk (int): The primary key of the Poll being updated.
+
+        Returns:
+            WSGIResponse: Upon success it redirects to "my-polls". Upon any failure
+            it returns to the poll editing page with the information pre-filled and any
+            error messages.
+        '''
+
+        # First get the data from the post and sort it out.
+        for k in ['question', 'open_date', 'close_date', 'suspended']:
+            try:
+                poll_data[k] = request.POST.get(k)
+            except NameError:
+                poll_data = {k: request.POST.get(k)}
+
+        for k in request.POST.keys():
+            if k.startswith('choice_text'):
+                # We have a choice
+                choice_id = k.split('.')[1]
+                choice_text = request.POST.get(k)
+                try:
+                    choices.append({'choice_text': choice_text, 'id': choice_id})
+                except NameError:
+                    choices = [{'choice_text': choice_text, 'id': choice_id}]
+
+        # Pass the data for the Poll into PollForm and validate it.
+        poll_form = PollUpdateForm(data=poll_data, instance=Poll.objects.get(pk=pk))
+        poll_valid = poll_form.is_valid()
+
+        # For each Choice pass the data into ChoiceForm and validate.
+
+        try:
+            choice_forms = []  # this is ugly but I will do it for now.
+            for choice in choices:
+                choice_form = ChoiceForm(data={'choice_text': choice['choice_text']},
+                                         instance=Choice.objects.get(pk=choice['id']))
+                choice_forms.append({'form': choice_form, 'id': choice['id']})
+        except NameError:
+            pass
+            # There are no choices for some reason.
```

We should report an error back to the user if there are no choices. There should be at least one choice for a poll.